### PR TITLE
Migrate db.java to kotlin

### PR DESCRIPTION
**Pull request by Arthur-Milchior**
Due to typing issue, it's based on top of #12015 and #12013.

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/libanki/DB.kt [hunk 1]
```diff
@@ -4,93 +4,55 @@
  * Copyright (c) 2009 Andrew <andrewdubya@gmail.com>                                    *
  * Copyright (c) 2011 Norbert Nagold <norbert.nagold@gmail.com>                         *
  * Copyright (c) 2018 Mike Hardy <mike@mikehardy.net>                                   *
- *                                                                                      *
+ * *
```
Comment (lukstbit) [inline AnkiDroid/src/main/java/com/ichi2/libanki/DB.kt:7]: Small format issue: align the `*` and `*/`.

File: AnkiDroid/src/main/java/com/ichi2/libanki/DB.kt [hunk 2]
```diff
@@ -171,198 +117,216 @@ public class DB {
...
+                }
+            } else {
+                Timber.w("Not in a transaction. Cannot end transaction.")
             }
-        } else {
-            Timber.w("Not in a transaction. Cannot end transaction.");
         }
     }
+
+    init {
```
Comment (lukstbit) [inline AnkiDroid/src/main/java/com/ichi2/libanki/DB.kt:328]: @KotlinCleanup("remove init by directly initializing the properties")

**Comment by dae** [comment]
I seem to recall finding DB.java difficult to convert to Kotlin when I tried it, so well done in pushing this through. :-)

**Review by lukstbit — CHANGES_REQUESTED** [review]
Looks ok with two small issues.

**Review by david-allison — CHANGES_REQUESTED** [review]
This contains unrelated changes, please only have the two commits

**Comment by Arthur-Milchior** [comment]
It was ready for review. Being clear that the the only interesting commit is the migration itself. 
I'd have make a PR with only those two commits, but it wouldn't compile without the dependencies being taken care of first.

If I put it as "draft", I'd assume nobody would review it. Which is clearly not what I wanted, since it could be reviewed independently

**Review by lukstbit — APPROVED** [review]

**Review by david-allison — APPROVED** [review]

**Pull request merged**